tests/jax/test_fused_attn.py

```diff
@@ -236,19 +286,13 @@ def test_forward(self):
         primitive_valid, primitive_invalid = jnp.split(primitive_out, (self.valid_len_q,), axis=1)
         reference_valid, _ = jnp.split(reference_out, (self.valid_len_q,), axis=1)
 
-        # Skip elementwise comparison when dropout enabled
-        if self.is_training and self.dropout_prob > 0.:
-            return
-
-        np.testing.assert_allclose(primitive_valid, reference_valid, atol=1e-2, rtol=1e-4)
-        np.testing.assert_allclose(primitive_invalid, jnp.zeros_like(primitive_invalid))
+        assert_allclose(primitive_valid, reference_valid, dtype=self.dtype)
+        assert_allclose(primitive_invalid, jnp.zeros_like(primitive_invalid), dtype=self.dtype)
 
     def test_backward(self):
         """
         Test value_and_grad with JIT, which includes both forward and backward
         """
-        if not self.is_training:
-            pytest.skip("Backward doesn't support inference")
```

Comment:
Could you insight me why `is_training` and `dropout_prob` can be dropped here?

Reply:
The pytest skip condition here means `test_backward()` never executes when `is_training = False`, so there's no point for `@pytest.mark.parametrize('is_training',...)` to be a general parameter for the entire test class. I moved it to `test_forward()` only instead to produce a cleaner pytest output with fewer skips.

I saw your point about the dropout though and I will be bringing that back to maintain the sanity check.